scripts/get-sources

```diff
@@ -43,102 +42,104 @@ source "$BUILDER_DIR/scripts/common"
     cd - >/dev/null
 else
-    rm -rf $REPO
+    rm -rf "$REPO"
     if [ "$(git rev-parse --is-shallow-repository)" == "true" ] && [ "$GIT_CLONE_FAST" != "1" ]; then
         GIT_OPTIONS+="--unshallow"
     fi
     print_headers
-    if ! git clone $GIT_OPTIONS -n -q -b $BRANCH "$GIT_URL" $REPO; then
+    if ! git clone $GIT_OPTIONS -n -q -b $BRANCH "$GIT_URL" "$REPO"; then
         if [ "$IGNORE_MISSING" == "1" ]; then exit 0; else exit 1; fi
     fi
-    VERIFY_REF=HEAD
-    fresh_clone=1
+    VERIFY_REF=HEAD fresh_clone=:
 fi
 
-verify=true
-if [ "$NO_CHECK" == "1" ] || elementIn "$COMPONENT" ${NO_CHECK[@]}; then
-    echo "--> $COMPONENT has NO_CHECK enabled"
-    echo "--> NOT Verifying tags..."
-    verify=false
-fi
-
-if [ "$verify" == "true" ]; then
-    echo "--> Verifying tags..."
-    KEYRING_DIR_GIT="$BUILDER_DIR/keyrings/git/$(basename "$REPO")"
-    export KEYRING_DIR_GIT
-    if ! "$(dirname "$0")/verify-git-tag" $REPO $VERIFY_REF; then
-        # if verfication failed, remove fetched content to make sure we'll not
-        # use it
-        if [ "$fresh_clone" -eq 1 ]; then
-            rm -rf "$REPO"
-        else
-            rm -f "$REPO"/.git/FETCH_HEAD
-        fi
-        exit 1
+echo "--> Verifying tags..."
+KEYRING_DIR_GIT="$BUILDER_DIR/keyrings/git/${REPO##*/}"
+export KEYRING_DIR_GIT
+if ! { VERIFY_REF=$(git -C "$REPO" rev-parse -q --verify "$VERIFY_REF") &&
+    "$BUILDER_DIR/scripts/verify-git-tag" "$REPO" "$VERIFY_REF"; } then
+    # if verfication failed, remove fetched content to make sure we'll not
+    # use it
+    if "$fresh_clone"; then
+        rm -rf "$REPO"
+    else
+        rm -f "$REPO/.git/FETCH_HEAD"
     fi
+    exit 1
 fi
 
-if [ "$FETCH_ONLY" == "1" ]; then
+if [ "${FETCH_ONLY-}" == "1" ]; then
     exit 0
 fi
 
-CURRENT_BRANCH="$(cd $REPO; git branch | sed -n -e 's/^\* \(.*\)/\1/p')"
-if [ "$CURRENT_BRANCH" != "$BRANCH" ] || [ "$VERIFY_REF" == "HEAD" ]; then
-    pushd $REPO &> /dev/null
+CURRENT_BRANCH="$( git -C "$REPO" branch | sed -n -e 's/^\* \(.*\)/\1/p'; )"
+if [ "$CURRENT_BRANCH" != "$BRANCH" ] || [ "$VERIFY_REF" = "HEAD" ]; then
+    pushd "$REPO" &> /dev/null || exit 1
     red="$(echo -e '\033[1;31m')"
     green="$(echo -e '\033[1;32m')"
     normal="$(echo -e '\033[0;0m')"
-    if [ -n "$(git name-rev --name-only $BRANCH 2> /dev/null)" ]; then
+    if [ -n "$(git name-rev --name-only "$BRANCH" 2> /dev/null)" ]; then
         echo "--> Switching branch from $CURRENT_BRANCH branch to ${green}$BRANCH${normal}"
-        git checkout $BRANCH || exit 1
+        git branch -f "$BRANCH" "$VERIFY_REF" &&
+        git checkout "$BRANCH" || exit 1
     else
         echo -e "--> Switching branch from $CURRENT_BRANCH branch to new ${red}$BRANCH${normal}"
-        git checkout FETCH_HEAD -b $BRANCH || exit 1
+        git checkout "$VERIFY_REF" -b $BRANCH || exit 1
     fi
-    popd &> /dev/null
+    remote=$(git rev-parse --symbolic-full-name '@{u}') &&
+    [[ "$remote" =~ ^refs/remotes/ ]] &&
```

Comment:
Good point.  I will add `|| :` so that errors here are ignored, which is okay since they are basically harmless.  I will leave the actual code, though; it makes the output of `git log` much, *much* nicer.